subgraph.template.yaml

```diff
@@ -2,6 +2,74 @@ specVersion: 0.0.2
 schema:
   file: ./schema.graphql
 dataSources:
+  {{#iGenArt721CoreContractV3_Engine_FlexContracts}}
+  - kind: ethereum/contract
+    name: 'IGenArt721CoreV3_Engine_Flex{{#address}}-{{address}}{{/address}}'
+    network: {{network}}
+    source:
+      address: "{{#address}}{{address}}{{/address}}{{^address}}0x0000000000000000000000000000000000000000{{/address}}"
+      abi: IGenArt721CoreContractV3_Engine_Flex
+      {{#startBlock}}startBlock: {{startBlock}}{{/startBlock}}
+    mapping:
+      kind: ethereum/events
+      apiVersion: 0.0.5
+      language: wasm/assemblyscript
+      entities:
+        - Project
+        - ProjectScript
+        - Token
+        - Contract
+        - Account
+        - AccountProject
+        - Whitelisting
+        - Payment
+        - Sale
+        - SaleLookupTable
+        - ProposedArtistAddressesAndSplit
+      abis:
+        - name: IGenArt721CoreContractV3_Base
+          file: ./abis/IGenArt721CoreContractV3_Base.json
+        - name: IGenArt721CoreContractV3
+          file: ./abis/IGenArt721CoreContractV3.json
+        - name: IGenArt721CoreContractV3_Engine
+          file: ./abis/IGenArt721CoreContractV3_Engine.json
+        - name: IGenArt721CoreContractV3_Engine_Flex
+          file: ./abis/IGenArt721CoreContractV3_Engine_Flex.json
+        - name: GenArt721CoreV3_Engine_Flex
+          file: ./abis/GenArt721CoreV3_Engine_Flex.json
```

Comment:
you _might_ need to add these ABIs to the other V3 core dataSources. I'm not 100% sure, but I do know missing ABIs is not caught until runtime.

Reply:
yup, discussed with Ben offline but wanted to comment here for posterity. Everywhere we're doing the 
```
const flagshipContract = getV3FlagshipContract(event.address);
  if (flagshipContract) {
    ...
  }
  const engineContract = getV3EngineContract(event.address);
  if (engineContract) {
    ...
  }
```
we're potentially calling the GenArt721CoreV3_Engine_Flex contract so we need the abi. I think in practice we'd early return everywhere before hitting this unless it was an unknown contract but i think since it's potentially called in the same handler it makes sense to add the abi. Good catch!